### PR TITLE
Fix NetHook2, modernize C++

### PR DESCRIPTION
**Pull request by yaakov-h**
- Fixes an assertion failure / crash immediately after injecting
- Introduces modern C++ features to deal with some IDE warnings.

### CODE REVIEW

File: Resources/NetHook2/NetHook2/logger.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ CLogger::CLogger()
 	m_LogDir = ss.str();
 
 	// create the session log directory
-	CreateDirectoryA( m_LogDir.c_str(), NULL );
+	CreateDirectoryA( m_LogDir.c_str(), nullptr);
```
Comment (VoiDeD) [inline Resources/NetHook2/NetHook2/logger.cpp]: Nit: whitespace.

File: Resources/NetHook2/NetHook2/logger.cpp [hunk 2]
```diff
@@ -12,18 +12,18 @@
 #include "steammessages_base.pb.h"
 
 
-CLogger::CLogger()
+CLogger::CLogger() noexcept
 {
 	m_uiMsgNum = 0;
 	char tempName[ MAX_PATH ];
-	GetModuleFileName( NULL, tempName, MAX_PATH );
+	GetModuleFileName(nullptr, tempName, MAX_PATH );
```
Comment (VoiDeD) [inline Resources/NetHook2/NetHook2/logger.cpp]: Nit: whitespace.

File: Resources/NetHook2/NetHook2/crypto.cpp [hunk 3]
```diff
@@ -53,33 +41,34 @@ CCrypto::CCrypto()
...
+		constexpr uint32 uMessageListStartPtrOffset = 23;
+		constexpr uint32 uMessageListCountPtrOffset = 10;
 
 		MsgInfo_t *pInfos = *(MsgInfo_t **)( pGetMessageList + uMessageListStartPtrOffset );
 		const uint32 uNumMessages = *(uint32 *)( pGetMessageList + uMessageListCountPtrOffset );
 
 		g_pLogger->LogConsole( "pGetMessageList = 0x%x\npInfos = 0x%x\nnumMessages = %d\n", pGetMessageList, pInfos, uNumMessages );
 		
 		HMODULE sc = GetModuleHandle("steamclient.dll");
+		assert(sc != nullptr);
```
Comment (VoiDeD) [inline Resources/NetHook2/NetHook2/crypto.cpp]: Should this fail with an error message instead? Someone could be loading into a process that hasn't loaded steamclient yet.

File: Resources/NetHook2/NetHook2/logger.cpp [hunk 4]
```diff
@@ -93,10 +93,10 @@ void CLogger::LogSessionData( ENetDirection eDirection, const uint8 *pData, uint
 
 	std::string fullFileTmp = fullFile + ".tmp";
 	std::string fullFileFinal = fullFile + ".bin";
-	HANDLE hFile = CreateFile( fullFileTmp.c_str(), GENERIC_WRITE, FILE_SHARE_READ, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL );
+	HANDLE hFile = CreateFile( fullFileTmp.c_str(), GENERIC_WRITE, FILE_SHARE_READ, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
```
Comment (VoiDeD) [inline Resources/NetHook2/NetHook2/logger.cpp]: Nit: whitespace.

File: Resources/NetHook2/NetHook2/logger.cpp [hunk 5]
```diff
@@ -93,10 +93,10 @@ void CLogger::LogSessionData( ENetDirection eDirection, const uint8 *pData, uint
 
 	std::string fullFileTmp = fullFile + ".tmp";
 	std::string fullFileFinal = fullFile + ".bin";
-	HANDLE hFile = CreateFile( fullFileTmp.c_str(), GENERIC_WRITE, FILE_SHARE_READ, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL );
+	HANDLE hFile = CreateFile( fullFileTmp.c_str(), GENERIC_WRITE, FILE_SHARE_READ, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
 
 	DWORD numBytes = 0;
-	WriteFile( hFile, pData, cubData, &numBytes, NULL );
+	WriteFile( hFile, pData, cubData, &numBytes, nullptr);
```
Comment (VoiDeD) [inline Resources/NetHook2/NetHook2/logger.cpp]: Nit: whitespace.

File: Resources/NetHook2/NetHook2/logger.cpp [hunk 6]
```diff
@@ -110,10 +110,10 @@ HANDLE CLogger::OpenFile( const char *szFileName, bool bSession )
 	std::string outputFile = ( bSession ? m_LogDir : m_RootDir );
 	outputFile += szFileName;
 
-	return CreateFile( outputFile.c_str(), GENERIC_WRITE, FILE_SHARE_READ, NULL, OPEN_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL );
+	return CreateFile( outputFile.c_str(), GENERIC_WRITE, FILE_SHARE_READ, nullptr, OPEN_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
```
Comment (VoiDeD) [inline Resources/NetHook2/NetHook2/logger.cpp]: Nit: whitespace.

File: Resources/NetHook2/NetHook2/logger.cpp [hunk 7]
```diff
@@ -131,14 +131,14 @@ void CLogger::LogOpenFile( HANDLE hFile, const char *szFmt, ... )
...
+	const int len = vsprintf_s( szBuff, buffSize, szFmt, args );
 
 	szBuff[ buffSize - 1 ] = 0;
 
-	SetFilePointer( hFile, 0, NULL, FILE_END );
+	SetFilePointer( hFile, 0, nullptr, FILE_END );
 
 	DWORD numBytes = 0;
-	WriteFile( hFile, szBuff, len, &numBytes, NULL );
+	WriteFile( hFile, szBuff, len, &numBytes, nullptr);
```
Comment (VoiDeD) [inline Resources/NetHook2/NetHook2/logger.cpp]: Nit: whitespace.

File: Resources/NetHook2/NetHook2/crypto.cpp (hunk 3 above)
Comment (yaakov-h) [reply Resources/NetHook2/NetHook2/crypto.cpp]: Our own injector checks for it pre-injection and has error handling for this already - see `k_ESteamProcessSearchErrorTargetProcessDoesNotHaveSteamClientDllLoaded`.

**Pull request merged**